src/target/riscv/riscv-013.c

```diff
@@ -2831,33 +2839,50 @@ static int read_memory_bus_v1(struct target *target, target_addr_t address,
 						return ERROR_FAIL;
 				}
 				if (next_read != address - 1) {
-					buf_set_u32(buffer + next_read - address, 0, 8 * MIN(size, 4), value);
-					log_memory_access(next_read, value, MIN(size, 4), true);
+					buf_set_u32(buffer + next_read - address, 0, 8 * MIN(size, 4), sbvalue[next_read_j]);
+					if (next_read_j == 0) {
+						if (size == 16) {
+							uint64_t value_h = ((uint64_t)sbvalue[3] << 32) | sbvalue[2];
+							uint64_t value_l = ((uint64_t)sbvalue[1] << 32) | sbvalue[0];
+							log_memory_access128(next_read, value_h, value_l, true);
+						} else {
+							uint64_t value = ((uint64_t)sbvalue[1] << 32) | sbvalue[0];
+							log_memory_access(next_read, value, size, true);
+						}
+						memset(sbvalue, 0, size);
```

Comment:
I'm not sure whether to keep this memset, can anyone give me some advice

Reply:
Keep it. If it turns out to slow things down (I doubt it), then we can put this whole block inside a check so it only runs when debug is enabled.